src/components/QueryEditor.tsx

```diff
@@ -598,6 +596,123 @@ const QueryEditor: QueryEditorComponent = ({
       ).concat(DEFAULT_RETURN_NODE),
     [viewConditions]
   );
+  useEffect(() => {
+    // Create Initial Branch and Condition
+    if (view.uid === parentUid) return;
+
+    const branches = viewCondition?.conditions.length;
+    const branchUid = getNthChildUidByBlockUid({
+      blockUid: view.uid,
+      order: view.branch,
+    });
+    const branchConditions = getChildrenLengthByPageUid(branchUid);
+
+    if (!branches) createBranch();
```

Comment:
[nit] we should be memoizing methods that enter a `useEffect` with `useCallback`s and making sure they are in the dependency array. If we are worried about `createBranch` updating too much, we could set it to a ref and access the ref inner value from the use effect

Reply:
so both this and `addCondition`